client/app/collections/contacts.coffee

```diff
@@ -1,32 +1,45 @@
 ContactsListener = require 'lib/contacts_listener'
+Contact = require 'models/contact'
 
 
-comparator = (sort) ->
-    fn = (a, b) ->
-        [a, b] = [a, b].map (model) ->
-            name = model.get('n').split(';')
-            if fn.sort is 'gn'
-                "#{name[0]}#{name[1]}"
+
+module.exports = class Contacts extends Backbone.Collection
+
+    model: Contact
+    url: 'contacts'
+
+    sortDisabled: false
+
+    buildComparator: (sort) =>
+        fn = (a, b) =>
+            if @sortDisabled
+                return -1
             else
-                "#{name[1]}#{name[0]}"
+                [a, b] = [a, b].map (model) ->
```

Comment:
It needs to be tested of course, but something along the line of :
```coffee
(a, b) ->
    if fn.sort is 'gn'
         a.get('n').localCompare b.get('n') # no need to split & join if we have the same order
   else
        a.getLastFirstName().localCompare b.getLastFirstName()
```

Should be faster.